aerogear/mobile-developer-console-operator
# ✨ Create mobile-developer Role/Binding in controller

JIRA: https://issues.jboss.org/browse/AEROGEAR-9996

While in OpenShift v3.11 we can create this role and rolebinding as
normal using the yaml file, it's not possible to do through the OLM
CSV, because the subject of the rolebinding is a group rather than a
ServiceAccount.

I am actually not sure if putting it here in the operator is the most
appropriate place at all, since I think whoever creates the project
should do it, but this is at least a start.

Note that the operator ServiceAccount has been given extra permissions
as part of this, so updating to this version will involve re-applying
the deploy/role.yaml before the deploy/operator.yaml

I've also changed the name of the mobile-developer rolebinding from
mobile-developer-binding to mobile-developer, because there's no need
for the type to be indicated in the name of an instance of that
type. This will mean that existing installations will end up with 2
identical rolebindings with different names, and this isn't a
problem. Since the operator didn't create the first
one (mobile-developer-binding), I don't think it's appropriate for it
to delete it (and that would just be adding additional complexity for
no noticable gain). Once the new one is created, people are free to
delete the old one from their projects.

----

## Verification steps

Deploy the operator, make sure there is no mobile-developer role or rolebinding. Create an MDC CR, there should now be a mobile-developer role and rolebinding. Ensure there are no rbac-related errors in the operator logs, or generally no persistent errors.
